kernel/kernel-api/src/main/java/io/delta/kernel/Transaction.java

```diff
@@ -197,7 +197,7 @@ static CloseableIterator<FilteredColumnarBatch> transformLogicalData(
           }
 
           ColumnarBatch data = filteredBatch.getData();
-          if (!data.getSchema().equals(tableSchema)) {
+          if (!data.getSchema().equivalentIgnoreCollations(tableSchema)) {
```

Comment:
Can you explain to me when we want to use each of these? And how you made sure we fixed any of these issues in the src code base?